File: debug-export/dart/core/api_response.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

/// Normalized API Response
class ApiResponse<T> {
  final bool success;
  final String message;
  final T? data;
  final int? statusCode;

  ApiResponse({
    required this.success,
    required this.message,
    this.data,
    this.statusCode,
  });

  /// Create ApiResponse from HTTP response
  static ApiResponse<T> fromResponse<T>(
    http.Response response,
    T Function(dynamic)? dataParser,
  ) {
    try {
      final json = jsonDecode(response.body) as Map<String, dynamic>;
      
      // Extract fields based on response mapping
      final success = json['success'] as bool? ?? (response.statusCode >= 200 && response.statusCode < 300);
      final message = json['message'] as String? ?? '';
      final rawData = json['data'];
      final statusCode = json['statusCode'] as int? ?? response.statusCode;

      T? parsedData;
      if (rawData != null && dataParser != null) {
        parsedData = dataParser(rawData);
      }

      return ApiResponse<T>(
        success: success,
        message: message,
        data: parsedData,
        statusCode: statusCode,
      );
    } catch (e) {
      return ApiResponse<T>(
        success: false,
        message: 'Failed to parse response: $e',
        statusCode: response.statusCode,
      );
    }
  }

  @override
  String toString() {
    return 'ApiResponse(success: $success, message: $message, data: $data, statusCode: $statusCode)';
  }
}

File: final-test/typescript/core/api_response.ts
import { AxiosResponse } from 'axios';

/**
 * Normalized API Response
 */
export class ApiResponse<T = any> {
  success: boolean;
  message: string;
  data?: T;
  statusCode?: number;

  constructor({ success, message, data, statusCode }: {
    success: boolean;
    message: string;
    data?: T;
    statusCode?: number;
  }) {
    this.success = success;
    this.message = message;
    this.data = data;
    this.statusCode = statusCode;
  }

  toString(): string {
    return `ApiResponse(success: ${this.success}, message: ${this.message}, statusCode: ${this.statusCode})`;
  }
}

/**
 * Normalize axios response to ApiResponse format
 */
export function normalizeResponse<T = any>(response: AxiosResponse): ApiResponse<T> {
  try {
    const data = response.data;
    
    // Extract fields based on response mapping
    const success = data.success !== undefined 
      ? data.success 
      : (response.status >= 200 && response.status < 300);
    const message = data.message || '';
    const responseData = data.data !== undefined ? data.data : data;
    const statusCode = data.statusCode || response.status;

    return new ApiResponse<T>({
      success,
      message,
      data: responseData,
      statusCode,
    });
  } catch (error) {
    return new ApiResponse<T>({
      success: false,
      message: `Failed to parse response: ${error}`,
      data: undefined,
      statusCode: response.status,
    });
  }
}
